daemon/src/forward_only_ok.rs

```diff
@@ -0,0 +1,58 @@
+use std::fmt;
+
+use xtra::{Address, Handler, KeepRunning};
+
+/// A forwarding actor that only forwards [`Result::Ok`] values and shuts itself down upon the first
+/// error.
+pub struct Actor<T> {
+    forward: Address<T>,
+}
+
+impl<T> Actor<T> {
+    pub fn new(forward: Address<T>) -> Self {
+        Self { forward }
+    }
+}
+
+pub struct Message<TOk, TErr>(pub Result<TOk, TErr>);
+
+impl<TOk, TErr> xtra::Message for Message<TOk, TErr>
+where
+    TOk: Send + 'static,
+    TErr: Send + 'static,
+{
+    type Result = KeepRunning;
+}
+
+#[async_trait::async_trait]
+impl<T, TOk, TErr> Handler<Message<TOk, TErr>> for Actor<T>
+where
+    T: Handler<TOk>,
+    TOk: xtra::Message<Result = ()> + Send + 'static,
+    TErr: fmt::Display + Send + 'static,
+{
+    async fn handle(
+        &mut self,
+        Message(result): Message<TOk, TErr>,
+        _: &mut xtra::Context<Self>,
+    ) -> KeepRunning {
+        let ok = match result {
+            Ok(ok) => ok,
+            Err(e) => {
+                tracing::error!("Stopping forwarding due to error: {}", e);
```

Comment:
Are you happy with losing the cause? 

```suggestion
                tracing::error!("Stopping forwarding due to error: {:#}", e);
```

Reply:
I don't mind using alternate display, will not make a difference for most error types. Rendering causes of errors using the _alternate_ sigil is a feature of `anyhow::Error`, not of the `std::error::Error` type.

So unless `TErr` is `anyhow::Error`, it won't make a difference.